tests/MongoDB.Driver.Tests/AsyncCursorTests.cs

```diff
@@ -31,6 +33,77 @@ namespace MongoDB.Driver.Tests
     public class AsyncCursorTests
     {
         //public methods
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Cursor_should_call_close_in_thread_safe_way(
+            [Values(false, true)] bool cappedCollection,
+            [Values(false, true)] bool async)
+        {
+            string testCollectionName = "test";
+            string testDatabaseName = "test";
+            var client = CreateClient();
+            DropCollection(client, testDatabaseName, testCollectionName);
+            var database = client.GetDatabase(testDatabaseName);
+
+            var createCollectionOptions = new CreateCollectionOptions();
+            if (cappedCollection)
+            {
+                createCollectionOptions.Capped = true;
+                createCollectionOptions.MaxSize = 1000;
+            }
+
+            database.CreateCollection(testCollectionName, createCollectionOptions);
+            var collection = database.GetCollection<BsonDocument>(testCollectionName);
+            collection.InsertOne(new BsonDocument());
+
+            var findOptions = new FindOptions()
+            {
+                BatchSize = 1
+            };
+            if (cappedCollection)
+            {
+                findOptions.CursorType = CursorType.TailableAwait;
+            }
+            var findResult = collection.Find(FilterDefinition<BsonDocument>.Empty, findOptions);
+            IAsyncCursor<BsonDocument> cursor;
+            if (async)
+            {
+                cursor = findResult
+                    .ToCursorAsync()
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                cursor = findResult.ToCursor();
+            }
+
+            var asyncCursor = cursor.As<AsyncCursor<BsonDocument>>();
+            var enumerable = asyncCursor
+                .ToEnumerable()
+                .GetEnumerator();
+
+            enumerable.MoveNext();
+
+            var thread = new Thread(
+                () =>
+                {
+                    Thread.Sleep(1000);
+                    asyncCursor.Close();
+                });
+            thread.Start();
```

Comment:
I don't understand why there is a sleep and why we are not waiting for the thread to complete before attempting to use the cursor that has been closed. Seems like this should be:

    var thread = new Thread(
        () =>
        {
            asyncCursor.Close();
        });
    thread.Start();
    thread.Join();

    var exception = Record.Exception(() => enumerator.MoveNext());

But even beyond that... why are we even testing a different thread closing the cursor when by design AsyncCursor is NOT thread safe and we make no claims that it is.